File: lib/data/file_manager.py
import os

class FileManager:
    def __init__(self, base_path="data/files"):
        self.base_path = base_path
        os.makedirs(base_path, exist_ok=True)

    def write(self, filename, data):
        with open(f"{self.base_path}/{filename}", "w", encoding="utf-8") as f:
            f.write(data)

    def read(self, filename):
        with open(f"{self.base_path}/{filename}", "r", encoding="utf-8") as f:
            return f.read()

    def append(self, filename, data):
        with open(f"{self.base_path}/{filename}", "a", encoding="utf-8") as f:
            f.write(data)

    def exists(self, filename):
        return os.path.exists(f"{self.base_path}/{filename}")
    

File: lib/data/file_manager.dart
import 'dart:io';
import 'dart:convert';

class FileManager {
  final String basePath;

  FileManager({this.basePath = 'files'}) {
    Directory(basePath).createSync(recursive: true);
  }

  Future<void> write(String filename, String data) async {
    final file = File('$basePath/$filename'); // هنا
    await file.writeAsString(data, encoding: utf8);
  }

  Future<String> read(String filename) async {
    final file = File('$basePath/$filename'); // هنا
    return await file.readAsString(encoding: utf8);
  }

  Future<void> append(String filename, String data) async {
    final file = File('$basePath/$filename'); // هنا
    await file.writeAsString(data, mode: FileMode.append, encoding: utf8);
  }

  bool exists(String filename) {
    return File('$basePath/$filename').existsSync(); // هنا
  }

  Future<void> delete(String filename) async {
    final file = File('$basePath/$filename'); // هنا
    if (await file.exists()) {
      await file.delete();
    }
  }
}
